lutron-aurora/src/read_responder.lua

```diff
@@ -0,0 +1,45 @@
+local zb_messages = require "st.zigbee.messages"
+local zcl_messages = require "st.zigbee.zcl"
+local zb_const = require "st.zigbee.constants"
+local data_types = require "st.zigbee.data_types"
+local read_attr_response = require "st.zigbee.zcl.global_commands.read_attribute_response"
+local Level = require "st.zigbee.zcl.clusters".Level
+local Status = require "st.zigbee.generated.types.ZclStatus"
+local Uint8 = require "st.zigbee.data_types.Uint8"
+
+--- The goal here is to respond to the device's `Read` message with the current
+--- value of the switchLevel propery.
+---
+--- note: this seems to not really help with the device losing its own state
+return function(device)
+  local current = device:get_latest_state("main", "switchLevel", "level", 0)
+  local adjusted = math.floor((255 - 2) * (current / 100)) + 2
+  local response = read_attr_response.ReadAttributeResponse({
+    read_attr_response.ReadAttributeResponseAttributeRecord(
+      Level.attributes.CurrentLevel.ID,
+      Status.SUCCESS,
+      Uint8.ID,
+      adjusted
+    )
+  })
+  local zclh = zcl_messages.ZclHeader({
+    cmd = data_types.ZCLCommandId(read_attr_response.ReadAttributeResponse.ID)
+  })
+  local addrh = zb_messages.AddressHeader(
+      zb_const.HUB.ADDR,
+      zb_const.HUB.ENDPOINT,
+      device:get_short_address(),
+      device:get_endpoint(Level.ID),
+      zb_const.HA_PROFILE_ID,
+      Level.ID
+  )
+  local message_body = zcl_messages.ZclMessageBody({
+    zcl_header = zclh,
+    zcl_body = response,
+  })
+  local tx = zb_messages.ZigbeeMessageTx({
+    address_header = addrh,
+    body = message_body,
+  })
+  device:send(tx)
```

Comment:
I was hoping that doing this would allow the device itself to send an accurate state, instead of treating it like a stateless dimmer where events are either < or > 3. I am now seeing this go out and maybe get acknowledged but not change in the device.

```shell
# received read request
< ZigbeeMessageRx || type: 0x00, < AddressHeader || src_addr: 0xF298, src_endpoint: 0x01, dest_addr: 0x0000, dest_endpoint: 0xFF, profile: 0x0104, cluster: Level >, lqi: 0xC4, rssi: -61, body_length: 0x0005, < ZCLMessageBody || < ZCLHeader || frame_ctrl: 0x10, seqno: 0x75, ZCLCommandId: 0x00 >, < ReadAttribute || AttributeId: 0x0000 > > >
# sent response
< ZigbeeMessageTx || Uint16: 0x0000, < AddressHeader || src_addr: 0x0000, src_endpoint: 0x01, dest_addr: 0xF298, dest_endpoint: 0x01, profile: 0x0104, cluster: Level >, < ZCLMessageBody || < ZCLHeader || frame_ctrl: 0x00, seqno: 0x00, ZCLCommandId: 0x01 >, < ReadAttributeResponse || < AttributeRecord || attr_id: 0x0000, ZclStatus: SUCCESS, DataType: Uint8, data: 0xE5 > > > >
# Level
< ZigbeeMessageRx || type: 0x00, < AddressHeader || src_addr: 0xF298, src_endpoint: 0x01, dest_addr: 0x0000, dest_endpoint: 0xFF, profile: 0x0104, cluster: Level >, lqi: 0xD4, rssi: -64, body_length: 0x0006, < ZCLMessageBody || < ZCLHeader || frame_ctrl: 0x11, seqno: 0x77, ZCLCommandId: 0x04 >, < MoveToLevelWithOnOff || level: 0x02, transition_time: 0x0002 > > >
# ack?
< ZigbeeMessageRx || type: 0x00, < AddressHeader || src_addr: 0xF298, src_endpoint: 0x01, dest_addr: 0x0000, dest_endpoint: 0x01, profile: 0x0104, cluster: Level >, lqi: 0xD6, rssi: -64, body_length: 0x0005, < ZCLMessageBody || < ZCLHeader || frame_ctrl: 0x08, seqno: 0x14, ZCLCommandId: 0x0B >, < DefaultResponse || cmd: 0x01, ZclStatus: SUCCESS > > >
```
It seems odd that we get the MoveToLevel event before the ack (assuming that it is an ack?)